test/main.cpp

```diff
@@ -357,6 +368,17 @@ EXPORT int main(int argc, char const* argv[])
 		if (filter && tests_to_run.count(_g_unit_tests[i].name) == 0)
 			continue;
 
+		std::snprintf(dir, sizeof(dir), "sub_%i", i);
+		std::string unit_dir = combine_path(current_working_directory(), dir);
```

Comment:
this looks questionable. you overwrite "dir" here (as far as I can tell) with the per-test sub directory, the you add that to the current working directory, then you set the new current working directory to that.

I just notice that you "cd .." at the end of the test. What would you think about saving the root test directory you generate above, and reuse that to append the per-test subdirectory to. that way each way through the loop you would just cd straight to the new subdirectory.